### DISCUSSION TITLE
Help with filtering based on Class value

### DISCUSSION THREAD

**Discussion by quantumgardener** [mdelobelle/metadatamenu#603] [General]
I want to restrict values in an `entity` field to only those pages which have the property of `Class: Organisation`. This will allow me to filter to valid organisations only.

I am using a Select field type with the dv query of:

> `dv.list(dv.pages().where(p => p.Class === "Organisation").file.link)`

In other words, set entity to be a link to a page that is a valid organisation.

In straight dataviewjs in a note it works, but when in a field it doesn't and all I get is a blank popup.

I don't know how to debug what's going wrong here.

**Comment by PepeTravesi** [comment]
Hi.
I have the same issue with queries in metadata fields.
There are several options which accpet 'dataview query optional' said the  plugin but Im not able to make it run.
for example

```dataviewjs
const currentFolder = dv.current().file.folder;
dv.pages()
  .where(p => 
    p.file.path.startsWith(currentFolder) && 
    Array.isArray(p.hasTipoNota) && 
    p.hasTipoNota.includes("MOC")
  )
```
  
  It works on the note but doesnt on the fileClass
  
  I love this plugin but it´s really hard to work with because it lacks of queries documentation
  
  @quantumgardener, you can check if the field is a string or an array.  I had problems with this in the past
